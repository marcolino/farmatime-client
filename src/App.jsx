import React, { useState } from "react";
import { ThemeProvider, CssBaseline, createTheme } from "@mui/material";
//import { MuiThemeProvider, CssBaseline, createMuiTheme } from 'material-ui/styles';
import themeLight from "./themes/themeLight";
import themeDark from "./themes/themeDark";
import Header from "./components/Header";
import Footer from "./components/Footer";
import ThemeToggle from "./components/ThemeToggle";

function App() {
  const [theme, setTheme] = useState(themeLight);

  const toggleTheme = () => {
    setTheme((prevTheme) => (prevTheme.palette.mode === "light" ? themeDark : themeLight));
  };

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Header isLoggedIn={true} />
      <ThemeToggle theme={theme} toggleTheme={toggleTheme} />
      <Footer />
    </ThemeProvider>
  );
}

export default App;