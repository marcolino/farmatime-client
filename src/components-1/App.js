import React, { useContext } from "react";
import { makeStyles } from "@material-ui/styles";
import { BrowserRouter, useLocation } from "react-router-dom";
import { ThemeProvider } from "@mui/material/styles";
import CssBaseline from "@mui/material/CssBaseline";
//import { useTranslation } from "react-i18next";
import { AuthProvider } from "../providers/AuthProvider";
import { ServiceProvider } from "../providers/ServiceProvider";
import { OnlineStatusProvider, OnlineStatusContext } from "../providers/OnlineStatusProvider";
import Banner from "./Banner";
import ClientInfoDisplay from "./ClientInfoDisplay";
import Header from "./Header";
import Routes from "./Routes";
import Footer from "./Footer";
import Loader from "./Loader";
import CookieBanner from "./CookieBanner";
import { ToastContainer, toast } from "./Toast";
// import PushNotifications from "../components/PushNotifications";
// import FloatingActionButton from "./FloatingActionButton";
import { isAuthLocation } from "../libs/Misc";
import theme from "../themes/default"; // here we choose the theme
import config from "../config";

function App() {
  
  return (
    <ThemeProvider theme={theme}>
      <AuthProvider>
        <OnlineStatusProvider>
          <ServiceProvider>
            <CssBaseline />
            <BrowserRouter>
              {config.mode.development && <Banner text="development" />}
              {config.mode.development && <ClientInfoDisplay />}
              <Loader />
              <Contents />
              <CookieBanner />
              {/* <FloatingActionButton/> */}
            </BrowserRouter>
            {/* <PushNotifications /> */}
          </ServiceProvider>
        </OnlineStatusProvider>
      </AuthProvider>
      <ToastContainer />
    </ThemeProvider>
  );
}


const useStyles = makeStyles(theme => ({
  contentsContainer: {
    position: "relative",
    minHeight: "100vh",
  },
  contentsWrap: props => ({
    paddingBottom: props.footerHeight,
  }),
  header: {
  },
  body: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "left",
    padding: theme.spacing(2),
  },
  footer: props => ({
    position: "absolute",
    bottom: "0.5rem",
    width: "100%",
    height: props.footerHeight,
  }),
}));

const Contents = () => {
  const location = useLocation();
  const isOnline = useContext(OnlineStatusContext);
	const classes = useStyles({footerHeight: isAuthLocation(location) ? 0 : config.ui.footerHeight }); // hide footer while in auth screens

  return (
    <div className={classes.contentsContainer}>
      <div className={classes.contentsWrap}>
        <div className={classes.header}>
          <Header />
        </div>
        <div className={classes.body}>
          <Routes />
        </div>
      </div>
      <div className={classes.footer}>
        <Footer isOnline={isOnline} />
      </div>
    </div>
  );
}

export default App;