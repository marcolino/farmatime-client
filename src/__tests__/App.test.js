import { render, screen, waitForElementToBeRemoved } from "@testing-library/react";
//import renderer from "react-test-renderer";
import App from "../components/App";

test("renders Home screen", async () => {
  const { container } = render(<App />);
  const spinner = await container.querySelector("svg"); // define the spinner selector
  if (spinner) await waitForElementToBeRemoved(spinner); // wait for the spinner to be removed from DOM  

  expect(screen.getAllByText(/Quiccasa/i)[0]).toBeInTheDocument(); // check the presence of text "Home" in the DOM

  //console.log('container:', container);

  expect(container).toMatchSnapshot();

  // jest.mock('react-i18next', () => ({
  //   useTranslation: () => ({t: key => key})
  // }));
});

// TODO: learn toMatchSnapshot() ...
