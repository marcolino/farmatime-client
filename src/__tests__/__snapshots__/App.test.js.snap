// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home screen 1`] = `
<div>
  <div />
  <div
    class="makeStyles-contentsContainer-1"
  >
    <div
      class="makeStyles-contentsWrap-2 makeStyles-contentsWrap-6"
    >
      <div
        class="makeStyles-header-3"
      >
        <header>
          <header
            class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-header-10 mui-fixed MuiPaper-elevation3"
          >
            <div
              class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
            >
              <a
                href="/"
              >
                <img
                  alt="icon"
                  class="makeStyles-logo-11"
                  height="30"
                  name="LogoMain"
                  size="30"
                  src="LogoMain.png"
                  width="30"
                />
              </a>
              <div
                class="makeStyles-title-15"
              >
                Quiccasa
              </div>
              <div>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-menuLink-12 MuiTypography-colorInherit"
                  href="/searches"
                >
                  <span
                    class="makeStyles-headerLabel-14"
                  >
                    Searches
                  </span>
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-menuLink-12 MuiTypography-colorInherit"
                  href="/articles"
                >
                  <span
                    class="makeStyles-headerLabel-14"
                  >
                    Articles
                  </span>
                </a>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Join !
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </header>
          <div
            class="makeStyles-offset-9"
          />
        </header>
      </div>
      <div
        class="makeStyles-body-4"
      >
        <div
          style="display: none;"
        />
      </div>
    </div>
    <div
      class="makeStyles-footer-5 makeStyles-footer-7"
    >
      <div
        class="MuiContainer-root makeStyles-footer-23 MuiContainer-maxWidthLg"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
        >
          <h6
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            quiccasa-client
             
             
            v
            0.1.77
             
             ~ 
            Â©
             
             
             
            2022
            , 
             
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
              href="https://www.sistemisolari.com"
            >
              Sistemi Solari
            </a>
            <span>
              â€ƒ
            </span>
            <span>
              ðŸ‡¬ðŸ‡§
            </span>
            <span>
              â€ƒ
            </span>
            <img
              alt="Network connection indicator"
              fill="red"
              height="12"
              name="Network.on"
              size="12"
              src="Network.on.svg"
              title="Network connection is on"
              width="12"
            />
          </h6>
        </div>
      </div>
    </div>
  </div>
  <div
    class="CookieConsent"
    style="align-items: baseline; background: rgb(43, 55, 59); color: rgb(153, 153, 153); display: block; flex-wrap: wrap; justify-content: space-between; left: 0px; position: fixed; width: 100%; z-index: 999; font-family: Open Sans; bottom: 0px;"
  >
    <div
      class=""
      style="flex: 1 0 300px; margin: 15px;"
    >
      We use cookies or similar technologies as specified in the
       
       
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="/privacy-policy"
      >
        privacy policy
      </a>
      .
       
       
       
      You can consent to the use of such technologies by closing this notice
      ,
       
       
       
      or by otherwise continuing interacting with this site.
    </div>
    <div
      class=""
    >
      <button
        aria-label="Accept cookies"
        class=""
        id="rcc-confirm-button"
        style="background: rgb(255, 212, 45); border: 0px; border-radius: 0px; box-shadow: none; color: rgb(78, 80, 59); cursor: pointer; flex: 0 0 auto; padding: 5px 10px; margin: 15px; font-size: 13px; float: right;"
      >
        I agree
      </button>
    </div>
  </div>
</div>
`;
